guidelines/coding-guidelines.md

```diff
@@ -11,3 +11,59 @@ For public facing interfaces we favor read-only collections and enumerables. For
 * Avoid using `System.Linq`
 * Avoid using `foreach` over collections that do not have a struct enumerator
 
+## Dependencies
+
+### 3rd party dependencies
+
+#### Package dependencies
+
+We want to compile against the lowest major to catch breaking changes
+
+PackageReference: `[Major.0.0, Major+1)`
+
+#### Tests
+
+We want to compile against the latest version to catch breaking changes and bugs. This is the likely version that our users would be on.
+
+PackageReference: `[Major.*, Major+1)`
```

Comment:
thinking about this one, as we don't need to provide a nuspec dependency range on tests, is there any reason we wouldn't just use `Major.*`?

Reply:
Wouldn't this break all support branches when we release Major+1?

Reply:
@andreasohlund well, that was my question. Does a package reference stating version `x.*` not imply that you're tied towards version `x` when resolving the nuget dependency?

I think the reason for the dependency range syntax is, because the nuspec by default takes the package dependency and states this as a package dependency without an upper bound, which means `>= x.y` (where y is * being resolved to latest available version).

Reply:
Talking to @timbussmann we decided to keep as is for now